src/hapi.js

```diff
@@ -21,44 +26,37 @@ function registerRoutes(app, { routes : r }, callback) {
 
 function registerPlugins(app, { plugins : p }, callback) {
     const plugins = p.slice();
+    const promises = [];
 
     plugins.forEach(({ register, options },index,array) => {
         let pluginName =
           (register.register.attributes.pkg !== undefined) ? register.register.attributes.pkg.name : register.register.attributes.name;
         log.info(`Registering plugin: ${pluginName}`);
+        // In hapi 16, calling register without a callback will return a promise
+        // In hapi 17, this is not true
+        promises.push(app.register({ register : register, options : options}));
```

Comment:
Here's the documentation for the different server.register signatures:
https://hapijs.com/api/16.6.3#serverregisterplugins-options-callback
https://hapijs.com/api/17.5.1#serverregisterplugins-options-callback